src/components/counter/counter.test.tsx

```diff
@@ -0,0 +1,49 @@
+import { describe, expect, test } from 'vitest';
+import { screen, render } from '@testing-library/react';
+import Counter from './counter';
+import user from '@testing-library/user-event';
+describe('Counter', () => {
+  test('check component renders correctly', () => {
+    render(<Counter initialValue={1} />);
+    const counterElement = screen.getByRole('heading', { level: 1 });
+    expect(counterElement).toBeInTheDocument();
+    //const incrementButton =
+  });
+
+  test('renders a count of 1', () => {
+    render(<Counter initialValue={1} />);
+    const counterElement = screen.getByRole('heading', { level: 1 });
+    expect(counterElement).toHaveTextContent('1');
+  });
+
+  test('renders a count of 2 after clicking on increment button', async () => {
```

Comment:
make the name of test more descriptive . Not 'renders count of' but 'increments count on lick'. 2 here is kinda magic number.